Added JVM Version Checker

DaemonUmbra opened the pull request:
This should hopefully allow us to get warnings out to users on old or incompatible JVMs

DaemonUmbra committed:
Added JVM Version Checker

This should hopefully allow us to get warnings out to users on old JVMs

DaemonUmbra committed:
Update CheckVersion.java

Forgot to disable test functionality

diesieben07 commented:
Running `.java` files directly from the command line (without compilation) using the `java` executable was added in Java 11, so this feature can't be used here.

desht commented on server_files/run.sh:
```diff
@@ -1,6 +1,12 @@
+# Some versions of Java lack features necessary for Forge to function.
+# CheckVersion.java will be invoked to check for these versions.
+# To skip this check, simply comment out the following line:
+if ! java CheckVersion.java ; then exit 0; fi
```
This could be more concisely written as `java CheckVersion.java || exit 1` (diesieben's comment about running .java files directly notwithstanding)

Also note exit status 1, not 0 - non-zero exit code is Unix convention for non-successful script termination.

DaemonUmbra replied:
That was supposed to be a 1, oops

DaemonUmbra committed:
CheckVersion.java is now jvmchecker.jar

Due to the fact that Diesieben is correct about the java feature of running source code directly being a recent feature I've had to fall back to using a compiled jar with the version range set in the run scripts.
This PR has been checked with Java versions 8, 16, and 17.

DaemonUmbra commented:
This got too complex for what little it does to be accepted
